x/feedist/codec.proto

```diff
@@ -0,0 +1,71 @@
+
+
+// Revenue represents an account with funds collected from the fees. This is a
+// temporary account used for storing fees that are later distributed between
+// the owners.
+message Revenue {
+  // Admin key belongs to the governance entities. It can be used to transfer
+  // stored amount to an another account.
+  // While not enforced it is best to use a multisig contract here.
+  bytes admin = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+
+  // Recipients holds any number of addresses that the collected revenue is
+  // distributed to. This does not contain the weights required for the
+  // transfer operation, but declares addresses that must be involved in each
+  // funds distribution.
+  repeated Recipient recipients = 2;
+}
+
+message Recipient {
+  // An address that the funds should be transferred to.
+  // This should not be the validator addresses, as the keys used to sign
+  // blocks should never be in a wallet. This can be the wallets of the admins
+  // of the validators.
+  bytes address = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+
+  // Weight defines what part of the total revenue goes to this recipient.
+  // Each recipient receives part of the total revenue amount proportional to
+  // the weight. For example, if there are two recipients with weights 1 and 2
+  // accordingly, distribution will be 1/3 to the first address and 2/3 to the
+  // second one.
+  int32 weight = 2;
+}
+
+// NewRevenueMsg is issuing the creation of a new revenue stream instance.
+message NewRevenueMsg {
+  // Admin key belongs to the governance entities. It can be used to transfer
+  // stored amount to an another account.
+  // While not enforced it is best to use a multisig contract here.
+  bytes admin = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+
+  // Recipients holds any number of addresses that the collected revenue is
+  // distributed to. This does not contain the weights required for the
+  // transfer operation, but declares addresses that must be involved in each
+  // funds distribution.
+  repeated Recipient recipients = 2;
+}
+
+// DistributeMsg is a request to distribute all funds collected within a single
+// revenue instance. Revenue is distributed between recipients. Request must be
+// signed using admin key.
+message DistributeMsg {
+  // Revenue ID reference an ID of a revenue instance that the collected fees
+  // should be distributed between recipients.
+  bytes revenue_id = 1 [(gogoproto.customname) = "RevenueID"];
+}
+
+// UpdateRevenueMsg change the configuration of a revenue instance.
+// To assure recipients that they will receive money, every revenue update is
+// forcing funds distribution. Before applying any change all funds stored by
+// the revenue account are distributed using old configuration. Only when the
+// collected revenue amount is equal to zero the change is applied.
+message UpdateRevenueMsg {
```

Comment:
Good point with the name. I like the `ResetRevenueMsg`, because it sounds like the state will be cleaned up (funds are distributed). Unless someone will say no, I will change it before finishing this pull request.

Reply:
Done in https://github.com/iov-one/weave/pull/316/commits/25eb80cbef2faef866985901db0cec2140bac256